src/guc.c

```diff
@@ -477,10 +477,9 @@ _guc_init(void)
 							 NULL);
 
 	DefineCustomBoolVariable(MAKE_EXTOPTION("enable_decompression_logrep_markers"),
```

Comment:
At the moment, decompression markers sound specific to decompression activity. With this PR, the markers will be available for recompression as well. So, I think we should generalize the GUC to be compression markers (representing both decompression and recompression).

```suggestion
	DefineCustomBoolVariable(MAKE_EXTOPTION("enable_compression_logrep_markers"),
```

Reply:
I had considered that, but it would break compatibility with applications which are already using this guc. The way I justified retaining the name is to imagine that there are brackets around the "de" in decompression, like this: `enable_(de)compression_logrep_markers`.